dev/managedservicesplatform/internal/stack/monitoring/monitoring.go

```diff
@@ -0,0 +1,228 @@
+			Description: pointers.Ptr("Instance is taking a long time to start up - something may be blocking startup"),
+			ThresholdAggregation: &monitoringalertpolicy.ThresholdAggregation{
+				Filters:   map[string]string{"metric.type": "run.googleapis.com/container/startup_latencies"},
+				Aligner:   monitoringalertpolicy.MonitoringAlignPercentile99,
+				Reducer:   monitoringalertpolicy.MonitoringReduceMax,
+				Period:    "60s",
+				Threshold: 10000,
+			},
+		},
+	} {
+
+		config.ProjectID = vars.ProjectID
+		config.ServiceName = vars.Service.ID
+		config.ServiceKind = pointers.Deref(vars.Service.Kind, "service")
+		_, err := monitoringalertpolicy.New(stack, id, &config)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func serviceAlerts(stack cdktf.TerraformStack, id resourceid.ID, vars Variables) error {
+	// Only provision if MaxCount is specified above 5
+	if pointers.Deref(vars.MaxCount, 0) > 5 {
+		_, err := monitoringalertpolicy.New(stack, id, &monitoringalertpolicy.Config{
+			ID:          "instance_count",
+			Name:        "Container Instance Count",
+			Description: pointers.Ptr("There are a lot of Cloud Run instances running - we may need to increase per-instance requests make make sure we won't hit the configured max instance count"),
+			ProjectID:   vars.ProjectID,
+			ServiceName: vars.Service.ID,
+			ServiceKind: spec.ServiceKindService,
+			ThresholdAggregation: &monitoringalertpolicy.ThresholdAggregation{
+				Filters: map[string]string{"metric.type": "run.googleapis.com/container/instance_count"},
+				Aligner: monitoringalertpolicy.MonitoringAlignMax,
+				Reducer: monitoringalertpolicy.MonitoringReduceMax,
+				Period:  "60s",
+			},
+		})
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func jobAlerts(stack cdktf.TerraformStack, id resourceid.ID, vars Variables) error {
+	// Alert whenever a Cloud Run Job fails
+	_, err := monitoringalertpolicy.New(stack, id, &monitoringalertpolicy.Config{
+		ID:          "job_failures",
+		Name:        "Cloud Run Job Failures",
+		Description: pointers.Ptr("Failed executions of Cloud Run Job"),
+		ProjectID:   vars.ProjectID,
+		ServiceName: vars.Service.ID,
+		ServiceKind: spec.ServiceKindJob,
+		ThresholdAggregation: &monitoringalertpolicy.ThresholdAggregation{
+			Filters: map[string]string{
+				"metric.type":          "run.googleapis.com/job/completed_task_attempt_count",
+				"metric.labels.result": "failed",
+			},
+			GroupByField: "metric.label.result",
+			Aligner:      monitoringalertpolicy.MonitoringAlignCount,
+			Reducer:      monitoringalertpolicy.MonitoringReduceSum,
+			Period:       "60s",
+			Threshold:    0,
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func responseCodeMetrics(stack cdktf.TerraformStack, id resourceid.ID, vars Variables) error {
+	for _, config := range vars.Monitoring.Alerts.ResponseCodeRatios {
+
+		_, err := monitoringalertpolicy.New(stack, id, &monitoringalertpolicy.Config{
```

Comment:
Every resource needs a unique ID - `monitoringalertpolicy` happens to only create a single resource that we give a unique sub-ID but I think it's good still to add an ID on top, e.g.

```suggestion
		_, err := monitoringalertpolicy.New(stack, id.Group(...), &monitoringalertpolicy.Config{
```

to ensure it's scoped